osf/external/gravy_valet/request_helpers.py

```diff
@@ -154,13 +192,159 @@ def _make_gv_request(
             requested_resource=requested_resource,
         )
     )
-    response = requests.get(endpoint_url, headers=auth_headers, params=params)
+    if request_method == 'POST':
+        auth_headers['Content-Type'] = 'application/vnd.api+json'
+        response = requests.post(endpoint_url, headers=auth_headers, json={'data': data})
+    else:
+        response = requests.get(endpoint_url, headers=auth_headers, params=params)
     if not response.ok:
         # log error to Sentry
         pass
     return response
 
 
+def get_citation_url_list(auth, addon_short_name, project, request=None, pid=None):
+    if pid:
+        project_url = settings.DOMAIN + pid
+    else:
+        project_url = settings.DOMAIN + request.view_args.get('pid')
+    resource_references_response = get_gv_result(
+        endpoint_url=RESOURCE_LIST_ENDPOINT,
+        requesting_user=auth.user,
+        requested_resource=project,
+        params={'filter[resource_uri]': project_url},
+    )
+    if not resource_references_response:
+        return []
+    configured_citation_addons_url = resource_references_response.get_related_link(
+        'configured_citation_addons')
+    addons_url_list = get_gv_result_json(
+        endpoint_url=configured_citation_addons_url,
+        requesting_user=auth.user,
+        requested_resource=project,
+        request_method='GET',
+        params={}
+    )
+    gv_addon_name = addon_short_name if addon_short_name != 'zotero' else 'zotero_org'
+    citation_url_list = list(
+        filter(lambda x: x['attributes']['external_service_name'] == gv_addon_name, addons_url_list))
+    return citation_url_list
+
+
+def _get_gv_response(auth, addon, project, list_id):
+    data = {
+        'attributes': {
+            'operation_name': 'list_root_collections',
```

Comment:
Can you elaborate on the reason why one may list root collections only?

Reply:
In 250 row I check if list_id exists and change to list_collection_items

Reply:
Can you do this either on 237 line please, or 264, but not on both, because this is confusing

Reply:
@Vlad0n20 
It'd be great if you fixed that one